engineering-health-check/cxInsight_9_0.ps1

```diff
@@ -70,47 +80,150 @@ function getOAuth2Token() {
 $token = getOAuth2Token
 
-function getOdata() {
-    $Url = "${cx_sast_server}/cxwebinterface/odata/v1/Scans?`$select=Id,ProjectName,OwningTeamId,TeamName,ProductVersion,EngineServerId,Origin,PresetName,ScanRequestedOn,QueuedOn,EngineStartedOn,EngineFinishedOn,ScanCompletedOn,ScanDuration,FileCount,LOC,FailedLOC,TotalVulnerabilities,High,Medium,Low,Info,IsIncremental,IsLocked,IsPublic&`$expand=ScannedLanguages(`$select=LanguageName)&`$filter=ScanRequestedOn%20gt%20${start_date}Z%20and%20ScanRequestedOn%20lt%20${end_date}z"
+function getScanOdata {
+    param (
+        $outputFile
+    )
+    Write-Verbose "Retrieving scan data"
+
+    $Url = "${cx_sast_server}/cxwebinterface/odata/v1/Scans?`$select=Id,ProjectId,OwningTeamId,TeamName,ProductVersion,EngineServerId,Origin,PresetName,ScanRequestedOn,QueuedOn,EngineStartedOn,EngineFinishedOn,ScanCompletedOn,ScanDuration,FileCount,LOC,FailedLOC,TotalVulnerabilities,High,Medium,Low,Info,IsIncremental,IsLocked,IsPublic&`$expand=ScannedLanguages(`$select=LanguageName)&`$filter=ScanRequestedOn%20gt%20${start_date}Z%20and%20ScanRequestedOn%20lt%20${end_date}z"
     $headers = @{
         Authorization = $token
     }
     try {
         if ($bypassProxy) {
-            $response = Invoke-RestMethod -noProxy -uri "$Url" -method get -headers $headers -OutFile "data.txt"
+            $response = Invoke-RestMethod -noProxy -uri "$Url" -method get -headers $headers -OutFile $outputFile
         }
         else {
-            $response = Invoke-RestMethod -uri "$Url" -method get -headers $headers -OutFile "data.txt"
+            $response = Invoke-RestMethod -uri "$Url" -method get -headers $headers -OutFile $outputFile
         }
-        return $response
     }
     catch {
         Write-Host "Exception:" $_
         Write-Host "StatusCode:" $_.Exception.Response.StatusCode.value__
         Write-Host "StatusDescription:" $_.Exception.Response.StatusDescription
-		Write-Host $Url
-		Write-Host "An Error has prevented this script from collecting Odata."
+	Write-Host $Url
+	Write-Host "An error has prevented this script from collecting scan Odata."
+        exit(-1)
+    }
+}
+
+function getResultOData {
+    param (
+        $outputFile
+    )
+    Write-Verbose "Retrieving result data"
+
+    $Url = "${cx_sast_server}/cxwebinterface/odata/v1/Projects?`$select=Id"
+    $headers = @{
+        Authorization = $token
+    }
+    try {
+        if ($bypassProxy) {
+            $response = Invoke-RestMethod -noProxy -uri "$Url" -method get -headers $headers
+        }
+        else {
+            $response = Invoke-RestMethod -uri "$Url" -method get -headers $headers
+        }
+
+        $states = @{
+            "0" = "To Verify"
+            "1" = "Not Explotable"
+            "2" = "Confirmed"
+            "3" = "Urgent"
+            "4" = "Proposed Not Exploitable"
+        }
+        $projects = @{}
+        $response | Select-Object -ExpandProperty Value | ForEach-Object {

```

Comment:
This loop might take a really long time to complete.  Write-Progress might be a good idea so the user knows it is running.

Not suggesting to do this now, but multithreading this loop might be something to do in the future.

Reply:
I already had a call to Write-Verbose in there but adding Write-Progress as well seems like a good idea.